src/extensions/reorder-columns/bootstrap-table-reorder-columns.js

```diff
@@ -136,7 +136,9 @@ $.BootstrapTable = class extends $.BootstrapTable {
         const sortOrder = {}
 
         table.el.find('th').each((i, el) => {
-          sortOrder[$(el).data('field')] = i
+          if ('field' in el.dataset) {
```

Comment:
> The [in operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in) can check if a given attribute exists: `'keyname' in element.dataset`. Note that this will walk the [prototype chain](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain) of dataset and may be unsafe if you have external code that may pollute the prototype chain. Several alternatives exist, such as [`Object.hasOwn(element.dataset, 'keyname')`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn), or just checking if `element.dataset.keyname !== undefined`.

URL: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset#accessing_values

```suggestion
          if (el.dataset.field !== undefined) {
            sortOrder[el.dataset.field] = i
          }
```